test/conformance/kernel/urKernelGetSuggestedLocalWorkSize.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (C) 2023 Intel Corporation
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+// See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <uur/fixtures.h>
+
+struct urKernelGetSuggestedLocalWorkSizeTest : uur::urKernelExecutionTest {
+    void SetUp() override {
+        program_name = "bar";
+        UUR_RETURN_ON_FATAL_FAILURE(urKernelExecutionTest::SetUp());
+    }
+    size_t global_size = 32;
+    size_t global_offset = 0;
+    size_t n_dimensions = 1;
+
+    size_t suggested_local_work_size;
+};
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urKernelGetSuggestedLocalWorkSizeTest);
+
+TEST_P(urKernelGetSuggestedLocalWorkSizeTest, Success) {
+    suggested_local_work_size = SIZE_MAX;
+    ASSERT_SUCCESS(urKernelGetSuggestedLocalWorkSize(
+        kernel, queue, n_dimensions, &global_offset, &global_size,
+        &suggested_local_work_size));
+    ASSERT_LE(suggested_local_work_size, global_size);
+}
+
+TEST_P(urKernelGetSuggestedLocalWorkSizeTest, Success2D) {
+    size_t global_size_2d[2] = {32, 32};
+    size_t global_offset_2d[2] = {0, 0};
+    size_t suggested_local_work_size_2d[2] = {SIZE_MAX, SIZE_MAX};
+    ASSERT_SUCCESS(urKernelGetSuggestedLocalWorkSize(
+        kernel, queue, 2, global_offset_2d, global_size_2d,
+        suggested_local_work_size_2d));
+    for (int I = 0; I < 2; ++I) {
+        ASSERT_LE(suggested_local_work_size_2d[I], global_size_2d[I]);
+    }
+}
+
+TEST_P(urKernelGetSuggestedLocalWorkSizeTest, Success3D) {
+    size_t global_size_3d[3] = {32, 32, 32};
+    size_t global_offset_3d[3] = {0, 0, 0};
+    size_t suggested_local_work_size_3d[3] = {SIZE_MAX, SIZE_MAX, SIZE_MAX};
+    ASSERT_SUCCESS(urKernelGetSuggestedLocalWorkSize(
+        kernel, queue, 3, global_offset_3d, global_size_3d,
+        suggested_local_work_size_3d));
+    for (int I = 0; I < 3; ++I) {
+        ASSERT_LE(suggested_local_work_size_3d[I], global_size_3d[I]);
+    }
+}
+
+TEST_P(urKernelGetSuggestedLocalWorkSizeTest, InvalidNullHandleKernel) {
+    ASSERT_EQ_RESULT(urKernelGetSuggestedLocalWorkSize(
+                         nullptr, queue, n_dimensions, &global_offset,
+                         &global_size, &suggested_local_work_size),
+                     UR_RESULT_ERROR_INVALID_NULL_HANDLE);
+}
+
+TEST_P(urKernelGetSuggestedLocalWorkSizeTest, InvalidNullHandleQueue) {
+    // LATER: some adapter's implementation of the API does not require queue,
+    // do we still need to test for it?
```

Comment:
We should keep this test unless we completely remove the queue argument.

Reply:
Sure, I will remove the comment then.